app/models/household_good.rb

```diff
@@ -0,0 +1,12 @@
+class HouseholdGood < ApplicationRecord
+  validates :weight, :category, presence: true
+  validates :name, presence: true, uniqueness: { scope: :category }
+
+  def key
+    @key ||= name.gsub(/[^0-9a-z]/i, ' ').strip.gsub(/\s+/, '_').downcase
```

Comment:
I _think_ you could use `parameterize` ([APIdock page](https://apidock.com/rails/String/parameterize)) here and get more or less the same results.

Reply:
oh cool :)